src/App.tsx

```diff
@@ -1,54 +1,89 @@
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 import './App.scss';
+import { PersonList } from './components/PersonList/PersonList';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const [isPersonListActive, setIsPersonListActive] = useState(false);
+
+  const applyQuery = useCallback(
```

Comment:
Take into account that `useCallback` don't affect on anything here.

Here described two cases where it make sense: https://react.dev/reference/react/useCallback#should-you-add-usecallback-everywhere